internal-server.md

```diff
@@ -160,7 +159,8 @@ hello zamboni, this is git@data running gitolite3 3.6.11-2 (Debian) on git 2.27.
 Once you have access you can:
 
 ```
-$ git annex sync --content
+$ git annex copy --to origin
+$ git push
```

Comment:
looks like something is missing before git push?
~~~
julien-macbook:~/data.neuro/basel-mp2rage $ git checkout -b working-branch
Switched to a new branch 'working-branch'
julien-macbook:~/data.neuro/basel-mp2rage $ touch julien-test.txt
julien-macbook:~/data.neuro/basel-mp2rage $ vi julien-test.txt 
julien-macbook:~/data.neuro/basel-mp2rage $ git add julien-test.txt 
julien-macbook:~/data.neuro/basel-mp2rage $ git commit 
[working-branch 09e9a2d] Added test file
 1 file changed, 1 insertion(+)
 create mode 100644 julien-test.txt
julien-macbook:~/data.neuro/basel-mp2rage $ 
julien-macbook:~/data.neuro/basel-mp2rage $ git-annex copy --to=origin
copy derivatives/labels/sub-P001/anat/sub-P001_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P002/anat/sub-P002_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P003/anat/sub-P003_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P004/anat/sub-P004_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P005/anat/sub-P005_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P006/anat/sub-P006_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P007/anat/sub-P007_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P010/anat/sub-P010_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P011/anat/sub-P011_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P012/anat/sub-P012_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P013/anat/sub-P013_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P014/anat/sub-P014_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P015/anat/sub-P015_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P016/anat/sub-P016_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P017/anat/sub-P017_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P019/anat/sub-P019_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P021/anat/sub-P021_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P022/anat/sub-P022_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P023/anat/sub-P023_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P024/anat/sub-P024_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P025/anat/sub-P025_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P026/anat/sub-P026_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P027/anat/sub-P027_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P028/anat/sub-P028_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P029/anat/sub-P029_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P030/anat/sub-P030_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P031/anat/sub-P031_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P033/anat/sub-P033_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P034/anat/sub-P034_UNIT1_lesion-manual.nii.gz ok
copy derivatives/labels/sub-P035/anat/sub-P035_UNIT1_lesion-manual.nii.gz ok
copy sub-P001/anat/sub-P001_UNIT1.nii.gz ok
copy sub-P002/anat/sub-P002_UNIT1.nii.gz ok
copy sub-P003/anat/sub-P003_UNIT1.nii.gz ok
copy sub-P004/anat/sub-P004_UNIT1.nii.gz ok
copy sub-P005/anat/sub-P005_UNIT1.nii.gz ok
copy sub-P006/anat/sub-P006_UNIT1.nii.gz ok
copy sub-P007/anat/sub-P007_UNIT1.nii.gz ok
copy sub-P010/anat/sub-P010_UNIT1.nii.gz ok
copy sub-P011/anat/sub-P011_UNIT1.nii.gz ok
copy sub-P012/anat/sub-P012_UNIT1.nii.gz ok
copy sub-P013/anat/sub-P013_UNIT1.nii.gz ok
copy sub-P014/anat/sub-P014_UNIT1.nii.gz ok
copy sub-P015/anat/sub-P015_UNIT1.nii.gz ok
copy sub-P016/anat/sub-P016_UNIT1.nii.gz ok
copy sub-P017/anat/sub-P017_UNIT1.nii.gz ok
copy sub-P019/anat/sub-P019_UNIT1.nii.gz ok
copy sub-P021/anat/sub-P021_UNIT1.nii.gz ok
copy sub-P022/anat/sub-P022_UNIT1.nii.gz ok
copy sub-P023/anat/sub-P023_UNIT1.nii.gz ok
copy sub-P024/anat/sub-P024_UNIT1.nii.gz ok
copy sub-P025/anat/sub-P025_UNIT1.nii.gz ok
copy sub-P026/anat/sub-P026_UNIT1.nii.gz ok
copy sub-P027/anat/sub-P027_UNIT1.nii.gz ok
copy sub-P028/anat/sub-P028_UNIT1.nii.gz ok
copy sub-P029/anat/sub-P029_UNIT1.nii.gz ok
copy sub-P030/anat/sub-P030_UNIT1.nii.gz ok
copy sub-P031/anat/sub-P031_UNIT1.nii.gz ok
copy sub-P033/anat/sub-P033_UNIT1.nii.gz ok
copy sub-P034/anat/sub-P034_UNIT1.nii.gz ok
copy sub-P035/anat/sub-P035_UNIT1.nii.gz ok
julien-macbook:~/data.neuro/basel-mp2rage $ git push
fatal: The current branch working-branch has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin working-branch
~~~

Reply:
I thought about this. I decided that it tells you if there's an issue and what to do about it (run `git push --set-upstream origin working-branch`) so it would be okay.

If we wrote that instruction explicitly in then I fear people would push from each of their `my-working-branch1` and `my-working-branch2` to `origin/working-branch`, or use the same branch name for all of their work, or mix up which branch goes where. A simple `git push` is the most reliable thing here.

The absolute most reliable option would be to document

```
git push -u origin $(git rev-parse --abbrev-ref HEAD)
```

but that still has the problem that if someone is using multiple remotes (e.g. they're syncing between `joplin` and `romane` and `data` in an ad-hoc way because it was faster at the time) then this might forcibly overwrite the setup they've created, and start sending everything back to `data`.

Do you have ways around this? I'm out of ideas.

Reply:
There's also this: https://davidwalsh.name/git-default-remote

```
git config --global push.default current
```

Reply:
i would suggest adding a comment like: "You might get the following message: "fatal: The current branch working-branch has no upstream branch.". If you do, follow the instructions (ie: run "git push --set-upstream...")"

Reply:
I handled this in https://github.com/neuropoly/data-management/commit/6fc063005716415564027c3b9045169ea60be7b4 by mentioning `push.default`. We can debate it more if you really want.